ext/fluxnet_simulations/data_processing.jl

```diff
@@ -1,12 +1,25 @@
 """
     var_missing(x; val = -9999)
 
-A function that checks if the value of `x` is equal to 
--9999, which is the value that Fluxnet uses for missing 
+A function that checks if the value of `x` is equal to
+-9999, which is the value that Fluxnet uses for missing
 data. Returns true if x == -9999.
 """
 var_missing(x; val = -9999) = x == val
 
+"""
+    hour_offset_to_period(hour_offset_from_UTC)
+
+Convert a hour offset (which may be fractional) to a Dates.Period
+that can be added to DateTime objects.
+Supports both integer and half-hour time zones (e.g., 5.5 hours).
+"""
+function hour_offset_to_period(hour_offset_from_UTC)
```

Comment:
This should be used throughout whenever we use hour_offset_from_UTC. there are examples where we still use the `hour_offset_from_UTC` directly and dont convert the fractional part